src/aqua/spell/trigger.aqua

```diff
@@ -31,7 +31,8 @@ func get_trigger(spell_id: string) -> ?TriggerEvent, ?Error:
 
     if s.success:
       if s.absent == false:
-        trigger <- TriggerEventJson.parse(s.str)
+        -- How to check that s.message is a valid TriggerEvent???
```

Comment:
Would be cool to allow putting there only some structure like `MailboxMessage` or smth which would include different types of events like `TriggerEvent` and other types of events or messages (for example, messages from the node for updating script or some KV values).

Sad that 1) aqua and marine don't support sum-types, 2) we don't know what we want from mailboxes and therefore we don't have any model for it.

Reply:
We should discuss it on a daily because it's a really tough question